### ISSUE TITLE
conditional inclusion / additional export level or field tag

### ISSUE THREAD

**Issue by kghenderson** [cue-lang/cue#2148]
i'm not sure what to call this or how to classify it, but i'll just describe the issue.

i have a directory with a large number of large cue files in several directories which are mainly filled with code and data, i.e. concrete/instance stuff and not definitions. 

one feature i love about cue is the nesting capability in that i can evaluate/export a single file, or a whole folder/package, or a parent folder and so on up the ladder. makes testing nice and performant that it can scale with what you're working on.

my problem is that because this project is so huge, cue is processing the whole thing every time (i get that's what it's _supposed_ to do...) but the thing is that i want the leaf level things to validate, and i want the data there locally so they can be exported to files at the low level, but when i include them into the parent structures i don't want to bring that low-level stuff with it. 

I know that i can use private/hidden objects to keep them in the package, but i want the low-level export to _include_ that data when I reference it, so I need them to be exported. maybe it's a concept of 'protected' but that doesn't seem quite right, just want them package exported, but not module exported. (sorry, naming things is sure hard, these names aren't right, just trying to see what works and what doesn't)

i do want to include particular fields like the names, descriptions, and object types for grand level validations, but i don't need that low-level detail in the grand/meta model. 

there might be a way to do this by convention, and maybe it'll be addressed when we can @embed and store that large data/code externally, and maybe filter out those values when you don't want them. or maybe an @local tag or something to leave it public in the package but not export it to other packages. 

i was also considering using exec to run shell commands/queries to import only values into the grand model, but this doesn't seem right either. 

bottom line is: i want cue to process that data, i don't want the cue to be slow to validate because it's trying to load all of these data files into memory when i'm evaluating the the entire model.

**Comment by mvdan** [comment]
We have `@embed` now :) And we have an issue to track alternative loading mechanisms so that you can configure how packages are loaded, rather than always loading ancestor packages with the same package name: https://github.com/cue-lang/cue/issues/2486

I believe these two resolve your request here; what do you think?

**Comment by mvdan** [comment]
Closing for now, but happy to reopen if you can share more info. Thanks!
